core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -387,6 +388,14 @@ private[spark] class MemoryStore(
 
   def remove(blockId: BlockId): Boolean = memoryManager.synchronized {
     val entry = entries.synchronized {
+      entries.get(blockId) match {
```

Comment:
Doesn't remove already return what was removed? if so you can avoid .get - just .remove here and match on it

Reply:
@srowen , thx, I am a little concerned if I call entries.remove firstly, will the returned obj already be cleared and it may encounter some reference issue or other unexpected cases? so I still remained using get here.

Reply:
Why would the objects be gone? remove just removes copies of references from the map; they still exist and won't go away if you keep the references to them here.